File: smartadapter/src/main/java/smartadapter/listener/OnItemSwipedListener.java
package smartadapter.listener;

/*
 * Created by Manne Öhlund on 2019-08-17.
 * Copyright (c) All rights reserved.
 */

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Listener for swipe callbacks from extensions of {@link smartadapter.widget.SwipeExtension}.
 */
public interface OnItemSwipedListener {

    /**
     * Notified from {@link smartadapter.widget.DragAndDropExtension#onSwiped(RecyclerView.ViewHolder, int)}
     * with target view holder and swipe direction.
     * @param viewHolder target view holder
     * @param direction direction the item was swiped
     */
    void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction);
}

File: smartadapter/src/main/java/smartadapter/listener/OnItemSwipedListener.kt
package smartadapter.listener

/*
 * Created by Manne Öhlund on 2019-08-17.
 * Copyright (c) All rights reserved.
 */

import androidx.recyclerview.widget.RecyclerView

typealias Direction = Int

/**
 * Listener for swipe callbacks from extensions of [smartadapter.widget.SwipeExtension].
 */
interface OnItemSwipedListener {

    /**
     * Notified from [smartadapter.widget.DragAndDropExtension.onSwiped]
     * with target view holder and swipe direction.
     * @param viewHolder target view holder
     * @param direction direction the item was swiped
     */
    fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Direction)
}

inline fun onItemSwipedListener(crossinline onSwiped: (
        viewHolder: RecyclerView.ViewHolder,
        direction: Direction) -> Unit) = object : OnItemSwipedListener {
    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Direction) {
        onSwiped(viewHolder, direction)
    }
}